src/binary/coreir.cpp

```diff
@@ -19,12 +21,59 @@
 using namespace std;
 using namespace CoreIR;
 
+
 string getExt(string s) {
   auto split = splitString<vector<string>>(s,'.');
   ASSERT(split.size()>=2,"File needs extention: " + s);
   return split[split.size()-1];
 }
 
+bool fileExists(string name) {
+  std:: ifstream infile(name);
+  return infile.good();
+}
+
+void parseLib(string lib,string& libname,string& filename,vector<string>& lpaths) {
+  struct utsname unameData;
+  uname(&unameData);
+  string osname = unameData.sysname;
+  cout << "osname " << osname << endl;
```

Comment:
Do we want to leave this in stdout? I would probably comment/use a debug macro that I could disable for releases